plasmapy/plasma/grids.py

```diff
@@ -1065,23 +1141,41 @@ def _validate(self):
                     f"grid: {self.units}."
                 )
 
+    @modify_docstring(prepend=AbstractGrid.volume_averaged_interpolator.__doc__)
     def volume_averaged_interpolator(
         self, pos: Union[np.ndarray, u.Quantity], *args, persistent=False
     ):
+        """
+        Notes
+        -----
+
+        This interpolator approximates the value of a quantity at a given
+        interpolation point using a weighted sum of the values at the eight grid
+        vertices that surround the point. The weighting factors are calculated by
+        defining a volume :math:`dx \\times dy \\times dz`
+        (where :math:`dx`, :math:`dy`, and :math:`dz` are the grid
+        spacings in each direction) around each grid vertex and around the
+        interpolation point. The contribution of each grid vertex is then
+        weighted by the fraction of the volume surrounding the interpolation
+        point that overlaps the volume surrounding that vertex. This effectively
+        introduces a linear interpolation between grid vertices.
+
+        This implementation of this algorithm assumes that the grid is uniformly
+        spaced and Cartesian.
+        """
```

Comment:
```suggestion
        """
        Notes
        -----

        This interpolator approximates the value of a quantity for all
        the desired interpolation points in ``pos``.  The value at a given
        point is approximated by taking a weighted average of values of
        the surrounding grid points.  The number of grid points used in
        the average depends on where the interpolated point is: 1 if
        the point is in a corner of the grid, 2 if the point is on an
        edge of the grid, 4 if the point is on a face of the grid, and
        8 if the point is in the volume of the grid.

        It is assumed the grid is regularly spaced along all axes, but the
        spacing along each axis (:math:`dx`, :math:`dy`, and :math:`dz`)
        need not be equal.  This grid spacing defines the imaginary volume
        :math:`dV = dx \\times dy \\times dy` centered on each grid point.

        **How the algorithm works**

        1. The 8, 4, 2, or 1 grid point(s) surrounding the point of interest
           is determined.
        2. The weights for each grid point is calculated by taking the
           imaginary volume :math:`dV` centered on the point of interested
           and determining the fraction of that volume that intersects with
           the volume of the surround grid point volumes.
        3. Use the weights determined in (2) and the associated grid quantities
           to calculate the weighted average.
        """
```

I felt this docstring was missing a few key points, so here's my stab at expanding it.  I feel it's a step in the right direction, but it is definitely not perfect.

Reply:
I think of the algorithm as always averaging over eight points, where the extra points outside of the grid just continue the grid spacing and have a value of zero?

Reply:
As I mentioned in https://github.com/PlasmaPy/PlasmaPy/pull/1173#issuecomment-892015118, this would be problematic for something like a DC magnetic field.  I think my suggested approach is a little safer, but I'm open to arguments against it.  Simulations are not my forté so I'm sure there's a better approach.

Also as mentioned in that comment, this could be a future PR where the user can defined how to treat the boundary.